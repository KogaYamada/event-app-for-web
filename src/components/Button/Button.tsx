import React, { FC } from 'react';
import styled from 'styled-components';
import { COLER } from '../../styles/color';

interface IButton {
    color?: string;
}

export const Button: FC<IButton> = ({ color }) => {
    const handleClick = () => {
        console.log('Click!');
    };

    return (
        <StyledButton color={color} onClick={handleClick}>
            Click!
        </StyledButton>
    );
};

const StyledButton = styled.button<IButton>`
    border-radius: 2px;
    background: ${({ color }) => color};
    color: ${COLER.RED};
`;
