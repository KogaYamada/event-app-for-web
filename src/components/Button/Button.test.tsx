import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Button } from './Button';

describe('button test', () => {
    let wrapper: ShallowWrapper<
        any,
        Readonly<{}>,
        React.Component<{}, {}, any>
    >;
    beforeEach(() => {
        wrapper = shallow(<Button />);
    });
    it('button snapshot', () => {
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('button color', () => {
        wrapper.setProps({ color: '#FF0000' });
        expect(wrapper.dive()).toMatchSnapshot();
    });
});
