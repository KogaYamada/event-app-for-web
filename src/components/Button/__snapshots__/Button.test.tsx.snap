// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test button color 1`] = `
<button
  className="sc-AxjAm efnSTE"
  color="#FF0000"
  onClick={[Function]}
>
  Click!
</button>
`;

exports[`button test button snapshot 1`] = `
<button
  className="sc-AxjAm AOlyX"
  onClick={[Function]}
>
  Click!
</button>
`;
